a1/src/main/java/ro/utcn/spet/a1/repository/api/TagRepository.java

```diff
@@ -0,0 +1,18 @@
+package ro.utcn.spet.a1.repository.api;
+
+import ro.utcn.spet.a1.model.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface TagRepository {
+    Tag save(Tag tag);
+
+    Optional<Tag> findById(int id);
+
+    void remove(Tag tag);
+
+    List<Tag> findAll();
+
+    Tag findByName(String name);
```

Comment:
When reading a single Tag, return an Optional instead of null when you don't find a corresponding entry (lie for the find by id).